data/web/json_api.php

```diff
@@ -1,1772 +1,1831 @@
+          process_edit_return(acl('edit', 'domainadmin', array_merge(array('username' => $items), $attr)));
+        break;
+        case "alias-domain":
+          process_edit_return(mailbox('edit', 'alias_domain', array_merge(array('alias_domain' => $items), $attr)));
+        break;
+        case "spam-score":
+          process_edit_return(mailbox('edit', 'spam_score', array_merge(array('username' => $items), $attr)));
+        break;
+        case "domain-admin":
+          process_edit_return(domain_admin('edit', array_merge(array('username' => $items), $attr)));
+        break;
+        case "admin":
+          process_edit_return(admin('edit', array_merge(array('username' => $items), $attr)));
+        break;
+        case "fwdhost":
+          process_edit_return(fwdhost('edit', array_merge(array('fwdhost' => $items), $attr)));
+        break;
+        case "fail2ban":
+          process_edit_return(fail2ban('edit', array_merge(array('network' => $items), $attr)));
+        break;
+        case "ui_texts":
+          process_edit_return(customize('edit', 'ui_texts', $attr));
+        break;
+        case "self":
+          if ($_SESSION['mailcow_cc_role'] == "domainadmin") {
+            process_edit_return(domain_admin('edit', $attr));
+          }
+          elseif ($_SESSION['mailcow_cc_role'] == "user") {
+            process_edit_return(edit_user_account($attr));
+          }
+        break;
+        // return no route found if no case is matched
+        default:
+          http_response_code(404);
+          echo json_encode(array(
+            'type' => 'error',
+            'msg' => 'route not found'
+          ));
+          exit();
+      }
+    break;
+    // return no route found if no case is matched
+    default:
+      http_response_code(404);
+      echo json_encode(array(
+        'type' => 'error',
+        'msg' => 'route not found'
+      ));
+      exit();
+  }
+}
+if ($_SESSION['mailcow_cc_api'] === true) {
+  if (isset($_SESSION['mailcow_cc_api']) && $_SESSION['mailcow_cc_api'] === true) {
+    unset($_SESSION['return']);
+  }
+}
+
+// Custom Code
+function check_empty_result($data, $object) {
+  if(empty($data)) {
+    http_response_code(404);
+    echo json_encode(array(
+        'type' => error_switch('error'),
+        'msg' =>  $object.' could not be found'
+    ));
+    exit();
+  }
+}
+
+function throw_not_configured($object) {
+  http_response_code(404);
+  echo json_encode(array(
+      'type' => error_switch('warning'),
+      'msg' =>  'No '.$object.' configured'
+  ));
+  exit();
+}
+
+function error_switch(string $level) {
+  $lvlint = 0;
+
+  switch($level) {
+    case 'info':
+      $lvlint = 0;
+      break;
+    case 'success':
+      $lvlint = 1;
+      break;
+    case 'warning':
+      $lvlint = 2;
+      break;
+    case 'danger':
+      $lvlint = 3;
+      break;
+    case 'error':
+      $lvlint = 4;
+      break;
+    default:
+      $lvlint = 0;
+      break;
+  }
+  return $lvlint;
+}
```

Comment:
What are the thoughts about returning an int for the type on messages?
That would make it easier to work with enums in other clients / languages.